src/k8s-extension/azext_k8s_extension/partner_extensions/ContainerInsights.py

```diff
@@ -98,18 +98,16 @@ def _invoke_deployment(cmd, resource_group_name, deployment_name, template, para
         logger.info(json.dumps(template, indent=2))
         logger.info('==== END TEMPLATE ====')
 
-    if cmd.supported_api_version(min_api='2019-10-01', resource_type=ResourceType.MGMT_RESOURCE_RESOURCES):
-        deployment_temp = cmd.get_models('Deployment', resource_type=ResourceType.MGMT_RESOURCE_RESOURCES)
-        deployment = deployment_temp(properties=properties)
-
-        if validate:
-            validation_poller = smc.validate(resource_group_name, deployment_name, deployment)
+    deployment_temp = cmd.get_models('Deployment', resource_type=ResourceType.MGMT_RESOURCE_RESOURCES)
+    deployment = deployment_temp(properties=properties)
+    if validate:
+        if cmd.supported_api_version(min_api='2019-10-01', resource_type=ResourceType.MGMT_RESOURCE_RESOURCES):
+            validation_poller = smc.begin_validate(resource_group_name, deployment_name, deployment)
             return LongRunningOperation(cmd.cli_ctx)(validation_poller)
-        return sdk_no_wait(no_wait, smc.create_or_update, resource_group_name, deployment_name, deployment)
+        else:
+            return smc.validate(resource_group_name, deployment_name, deployment)
 
-    if validate:
-        return smc.validate(resource_group_name, deployment_name, properties)
-    return sdk_no_wait(no_wait, smc.create_or_update, resource_group_name, deployment_name, properties)
+    return sdk_no_wait(no_wait, smc.begin_create_or_update, resource_group_name, deployment_name, deployment)

```

Comment:
This call in `begin_create_or_update` is failing. It is coming with `ERROR: 'DeploymentsOperations' object has no attribute 'begin_create_or_update'`

Reply:
As the upgrade of `azure-mgmt-resource` in CLI main repo has not been released, it can only be tested through the dev environment. So may I ask if you are using CLI in dev environment and pulled the latest code from remote dev branch?
The expected version of `azure-mgmt-resource` is `16.1.0`, please execute `pip list` to confirm whether the version is correct~

Reply:
@jonathan-innis May I ask can you pass the test now?